### PR TITLE
DOC: add remote upstream

### PR DESCRIPTION
**Pull request by rcomer**
## PR summary

The dev guide currently suggests that cloning your fork will automatically add the upstream remote, but I _think_ you have to add it manually.

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [X] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

**Comment by rcomer** [comment]
[Renders like this](https://output.circle-artifacts.com/output/job/88250a4d-f4a3-413b-bab1-fa483061b89c/artifacts/0/doc/build/html/devel/development_setup.html#retrieve-the-latest-version-of-the-code)

**Comment by rcomer** [comment]
"remote" is git's term and part of the command you need to use.  We have the link to GitHub's [Managing remote repositories](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories), which gives a bit more detail to what these things are, and I think the fact that you pass the GitHub url makes it fairly intuitive what you are linking to.  So I'm not sure what else this needs.

I do in general have a preference for brevity though, which I'm aware is just a preference.  So it's possible I'm just not seeing what you're seeing!

File: doc/devel/development_setup.rst [hunk 1]
```diff
@@ -59,13 +59,29 @@ should replace ``<your-username>`` with your GitHub username):
 
 
 This will place the sources in a directory :file:`matplotlib` below your
-current working directory, set up the ``origin`` remote to point to your own
-fork, and set up the ``upstream`` remote to point to the Matplotlib main
-repository (see also `Managing remote repositories <https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories>`__.)
-Change into this directory before continuing::
+current working directory and set up the ``origin`` remote to point to your own
```
Comment (timhoffm) [inline doc/devel/development_setup.rst:62]:
```suggestion
current working directory and set up a remote repository ``origin`` pointing to your own
```
Is this clearer?

**Comment by tacaswell** [comment]
I think the wording is fine as it is.

**Pull request merged**